google/cloud/spanner_dbapi/cursor.py

```diff
@@ -88,6 +89,20 @@ def execute(self, sql, args=None):
 
         self._res = None
 
+        if not self._connection.autocommit:
+            if (
+                not self.transaction
+                or self.transaction.committed
+                or self.transaction.rolled_back
+            ):
+                self.transaction = self._connection.session_checkout().transaction()
+                self.transaction.begin()
+                self._connection.transactions.append(self.transaction)
+
+            self._res = self.transaction.execute_sql(sql)
+            self._itr = PeekIterator(self._res)
+            return
```

Comment:
I no transaction started for this cursor, or the last transaction is commited/rollbacked, we're starting new transaction. If transaction is in progress, we're just running the given SQL in it.